app/src/main/java/com/example/event_finder_capstone/AddPhotoActivity.java

```diff
@@ -0,0 +1,165 @@
+        String description = etDescription.getText().toString();
+        String eventId = event.getId();
+        if (description.isEmpty()) {
+            Toast.makeText(AddPhotoActivity.this, "Description cannot be empty!", Toast.LENGTH_LONG).show();
+            pb.setVisibility(ProgressBar.INVISIBLE);
+            return;
+        }
+        if (photoFile == null || ivPostImage.getDrawable() == null) {
+            Toast.makeText(AddPhotoActivity.this, "Photo cannot be empty!", Toast.LENGTH_LONG).show();
+            pb.setVisibility(ProgressBar.INVISIBLE);
+            return;
+        }
+        ParseUser currentUser = ParseUser.getCurrentUser();
+        savePhoto(description, currentUser, photoFile, eventId);
+    }
+
+    private void savePhoto(String description, ParseUser currentUser, File photoFile, String eventId) {
+        Photo photo = new Photo();
+        photo.setEventId(event.getId());
+        photo.setDescription(description);
+        photo.setImage(new ParseFile(photoFile));
+        photo.setUser(currentUser);
+        photo.saveInBackground(e -> {
+            if (e != null) {
+                Toast.makeText(AddPhotoActivity.this, "Error saving post!", Toast.LENGTH_LONG).show();
+            }
+            Toast.makeText(AddPhotoActivity.this, "Posted!", Toast.LENGTH_LONG).show();
+            etDescription.setText("");
+            ivPostImage.setImageResource(0);
+            pb.setVisibility(ProgressBar.INVISIBLE);
+        });
+    }
+
+    private void launchCamera() {
+        Intent intent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
+        photoFile = getPhotoFileUri(photoFileName);
+        Uri fileProvider = FileProvider.getUriForFile(AddPhotoActivity.this, "com.codepath.photoprovider", photoFile);
+        intent.putExtra(MediaStore.EXTRA_OUTPUT, fileProvider);
+        if (intent.resolveActivity(getPackageManager()) != null) {
+            startActivityForResult(intent, CAPTURE_IMAGE_ACTIVITY_REQUEST_CODE);
+        }
+    }
+
+
+    private void resizePhoto(Bitmap resizedBitmap) {
+        ByteArrayOutputStream bytes = new ByteArrayOutputStream();
+        resizedBitmap.compress(Bitmap.CompressFormat.JPEG, PHOTO_QUALITY, bytes);
+        File resizedFile = getPhotoFileUri(photoFileName + "_resized");
+        try {
+            resizedFile.createNewFile();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        FileOutputStream fos = null;
+        try {
+            fos = new FileOutputStream(resizedFile);
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+        }
+        try {
+            assert fos != null;
+            fos.write(bytes.toByteArray());
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        try {
+            fos.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private File getPhotoFileUri(String photoFileName) {
```

Comment:
⚠️ Note: I think there are some opportunities to applying the naming we discussed earlier to clarify this method's functionality.